lib/active_merchant/billing/gateways/quickbooks.rb

```diff
@@ -51,7 +48,11 @@ class QuickbooksGateway < Gateway
       FRAUD_WARNING_CODES = ['PMT-1000', 'PMT-1001', 'PMT-1002', 'PMT-1003']
 
       def initialize(options = {})
-        requires!(options, :consumer_key, :consumer_secret, :access_token, :token_secret, :realm)
+        if options[:refresh_token]
```

Comment:
I know we generally avoid leaving comments in the source directly, but I'm wondering if this might be a good time to break that convention and leave a comment explicitly noting that the else condition is for backwards compatibility for OAuth 1? Took me a bit to realize that's what was going on.

Reply:
updated! let me know if that seems sufficient